Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1383,26 +1404,42 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
     auto oldStartPagePoint = std::exchange(m_selectionTrackingData.startPagePoint, IntPoint { [m_currentSelection firstCharCenter] });
     m_selectionTrackingData.selectionToExtendWith = WTFMove(m_currentSelection);
 
-    RetainPtr<PDFSelection> selection = [m_pdfDocument selectionFromPage:firstPageOfCurrentSelection atPoint:m_selectionTrackingData.startPagePoint toPage:m_documentLayout.pageAtIndex(oldStartPageIndex).get() atPoint:oldStartPagePoint];
+    RetainPtr selection = [m_pdfDocument selectionFromPage:firstPageOfCurrentSelection atPoint:m_selectionTrackingData.startPagePoint toPage:m_documentLayout.pageAtIndex(oldStartPageIndex).get() atPoint:oldStartPagePoint];
     [selection addSelection:m_selectionTrackingData.selectionToExtendWith.get()];
     setCurrentSelection(WTFMove(selection));
 }
 
-void UnifiedPDFPlugin::beginTrackingSelection(PDFDocumentLayout::PageIndex pageIndex, const WebCore::IntPoint& pagePoint, SelectionGranularity selectionGranularity, OptionSet<WebEventModifier> mouseEventModifiers)
+void UnifiedPDFPlugin::beginTrackingSelection(PDFDocumentLayout::PageIndex pageIndex, const WebCore::IntPoint& pagePoint, SelectionGranularity selectionGranularity, OptionSet<WebEventModifier> mouseEventModifiers, WebMouseEventButton mouseEventButton)
 {
     m_selectionTrackingData.granularity = selectionGranularity;
     m_selectionTrackingData.startPageIndex = pageIndex;
     m_selectionTrackingData.startPagePoint = pagePoint;
     m_selectionTrackingData.marqueeSelectionRect = { };
     m_selectionTrackingData.shouldMakeMarqueeSelection = mouseEventModifiers.contains(WebEventModifier::AltKey);
     m_selectionTrackingData.shouldExtendCurrentSelection = mouseEventModifiers.contains(WebEventModifier::ShiftKey);
+    m_selectionTrackingData.selectionToExtendWith = nullptr;
+    // We make a repaint request if we've started tracking selection again after a context menu interaction.
+    // This allows us to toggle back to an active appearance for the selections.
+    if (std::exchange(m_selectionTrackingData.lastHandledEventWasContextMenuEvent, false))
+        m_contentsLayer->setNeedsDisplay();
+
+    // Context menu events can only generate a word selection under the event, so we bail out of the rest of our selection tracking logic.
+    if (isContextMenuEvent(mouseEventButton, mouseEventModifiers))
+        return updateCurrentSelectionForContextMenuEventIfNeeded();
 
     if (m_selectionTrackingData.shouldExtendCurrentSelection)
         extendCurrentSelectionIfNeeded();
 
     continueTrackingSelection(pageIndex, pagePoint);
 }
 
+void UnifiedPDFPlugin::updateCurrentSelectionForContextMenuEventIfNeeded()
+{
+    auto page = m_documentLayout.pageAtIndex(m_selectionTrackingData.startPageIndex);
+    if (!m_currentSelection || !(IntRect([m_currentSelection boundsForPage:page.get()]).contains(m_selectionTrackingData.startPagePoint)))
+        setCurrentSelection([page selectionForWordAtPoint:m_selectionTrackingData.startPagePoint]);
```

Comment:
Is word granularity right, or does this need to use Lookup?

Reply:
Word granularity is wrong, but also not a relevant concept here, so I'll remove any changes to granularity tracking.

Reply:
Dropped in https://github.com/WebKit/WebKit/commit/969e56be3a56363a7cd3acee00649dbe12ef1814